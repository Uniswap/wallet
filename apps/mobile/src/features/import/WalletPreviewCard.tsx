import React from 'react'
import { TouchableArea } from 'src/components/buttons/TouchableArea'
import { SelectionCircle } from 'src/components/input/SelectionCircle'
import { Box, Flex } from 'src/components/layout'
import { Text } from 'src/components/Text'
import { Unicon } from 'src/components/unicons/Unicon'
import { ChainId } from 'src/constants/chains'
import { useIsDarkMode } from 'src/features/appearance/hooks'
import { useENS } from 'src/features/ens/useENS'
import { ElementName } from 'src/features/telemetry/constants'
import { shortenAddress } from 'src/utils/addresses'
import { formatUSDPrice, NumberType } from 'src/utils/format'

interface Props {
  address: string
  selected: boolean
  balance?: number | null
  onSelect: (address: string) => void
  name?: ElementName
  testID?: string
  hideSelectionCircle?: boolean
}

export const ADDRESS_WRAPPER_HEIGHT = 36
const UNICON_SIZE = 32

export default function WalletPreviewCard({
  address,
  selected,
  balance,
  onSelect,
  hideSelectionCircle,
  ...rest
}: Props): JSX.Element {
  const { name: ensName } = useENS(ChainId.Mainnet, address)
  const isDarkMode = useIsDarkMode()

  const unselectedBorderColor = isDarkMode ? 'none' : 'backgroundOutline'

  return (
    <TouchableArea
      backgroundColor="background1"
      borderColor={selected ? 'magentaVibrant' : unselectedBorderColor}
      borderRadius="rounded20"
      borderWidth={1}
      px="spacing16"
      py="spacing16"
      onPress={(): void => onSelect(address)}
      {...rest}>
      <Flex row alignItems="center" justifyContent="space-between">
        <Flex
          row
          alignItems="center"
          gap="spacing12"
          height={ADDRESS_WRAPPER_HEIGHT}
          justifyContent="flex-start">
          <Unicon address={address} size={UNICON_SIZE} />
          <Box>
            <Text variant="bodyLarge">{ensName ?? shortenAddress(address)}</Text>
            {balance ? (
              <Text color="textSecondary" variant="subheadSmall">
                {formatUSDPrice(balance, NumberType.FiatTokenQuantity)}
              </Text>
            ) : null}
          </Box>
        </Flex>
        {!hideSelectionCircle && (
          <SelectionCircle selected={selected} size="icon16" unselectedColor="textSecondary" />
        )}
      </Flex>
    </TouchableArea>
  )
}
