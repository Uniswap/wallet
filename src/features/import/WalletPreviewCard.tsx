import React from 'react'
import Check from 'src/assets/icons/check.svg'
import { AddressDisplay } from 'src/components/AddressDisplay'
import { TouchableArea } from 'src/components/buttons/TouchableArea'
import { Flex } from 'src/components/layout'
import { Text } from 'src/components/Text'
import { ElementName } from 'src/features/telemetry/constants'
import { formatUSDPrice, NumberType } from 'src/utils/format'

interface Props {
  address: string
  selected: boolean
  balance?: number | null
  onSelect: (address: string) => void
  name?: ElementName
  testID?: string
}

export const ADDRESS_WRAPPER_HEIGHT = 36
const UNICON_SIZE = 32
const CHECK_ICON_SIZE = 24

export default function WalletPreviewCard({
  address,
  selected,
  balance,
  onSelect,
  ...rest
}: Props): JSX.Element {
  return (
    <TouchableArea
      backgroundColor={selected ? 'background3' : 'background1'}
      borderColor={selected ? 'accentActive' : 'none'}
      borderRadius="rounded16"
      borderWidth={1}
      px="spacing16"
      py="spacing12"
      onPress={(): void => onSelect(address)}
      {...rest}>
      <Flex row alignItems="center" justifyContent="space-between">
        <Flex
          row
          alignItems="center"
          gap="spacing12"
          height={ADDRESS_WRAPPER_HEIGHT}
          justifyContent="flex-start">
          {selected && (
            <Flex centered height={UNICON_SIZE} width={UNICON_SIZE}>
              <Check height={CHECK_ICON_SIZE} width={CHECK_ICON_SIZE} />
            </Flex>
          )}
          <AddressDisplay
            address={address}
            showAccountIcon={!selected}
            size={UNICON_SIZE}
            textAlign="flex-start"
            variant="bodyLarge"
          />
        </Flex>
        {balance ? (
          <Text color="textSecondary" variant="subheadSmall">
            {formatUSDPrice(balance, NumberType.FiatTokenQuantity)}
          </Text>
        ) : null}
      </Flex>
    </TouchableArea>
  )
}
