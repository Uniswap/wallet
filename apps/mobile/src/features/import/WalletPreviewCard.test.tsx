import React from 'react'
import WalletPreviewCard from 'src/features/import/WalletPreviewCard'
import { ACCOUNT_ADDRESS_ONE } from 'src/test/fixtures'
import { render } from 'src/test/test-utils'

it('renders wallet preview card', () => {
  const tree = render(
    <WalletPreviewCard selected address={ACCOUNT_ADDRESS_ONE} onSelect={(): null => null} />
  )
  expect(tree).toMatchSnapshot()
})
