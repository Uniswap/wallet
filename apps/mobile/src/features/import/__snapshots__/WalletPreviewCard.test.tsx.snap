// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders wallet preview card 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 5,
      "left": 5,
      "right": 5,
      "top": 5,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#FC72FF",
      "borderRadius": 20,
      "borderWidth": 1,
      "display": "flex",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "height": 32,
              "width": 32,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "height": 32,
                "width": 32,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              clip={true}
              mask={
                <UniconMask
                  attributeData={
                    Object {
                      "0": "#5CFE9D",
                      "1": "#B8C3B7",
                      "2": Array [
                        Object {
                          "path": "M34 20C34 28.8366 26.8366 36 18 36C9.16344 36 2 28.8366 2 20L2 -3.8147e-06L34 -1.01717e-06V20Z",
                        },
                      ],
                      "3": Array [
                        Object {
                          "fillType": "evenOdd",
                          "path": "M16 8C12.2937 6.62854 9.37146 3.70632 8 0C6.62854 3.70632 3.70632 6.62854 0 8C3.70632 9.37146 6.62854 12.2937 8 16C9.37146 12.2937 12.2937 9.37146 16 8ZM8 11C9.65685 11 11 9.65685 11 8C11 6.34315 9.65685 5 8 5C6.34315 5 5 6.34315 5 8C5 9.65685 6.34315 11 8 11Z",
                        },
                      ],
                    }
                  }
                  showBorder={false}
                  size={32}
                />
              }
            >
              <View
                transform={
                  Array [
                    Object {
                      "scale": 0.8648648648648649,
                    },
                  ]
                }
              >
                <View
                  height={37}
                  width={37}
                  x={0}
                  y={0}
                >
                  <View
                    colors={
                      Array [
                        "#5CFE9D",
                        "#B8C3B7",
                      ]
                    }
                    end={null}
                    start={null}
                  />
                </View>
                <View
                  color="#EDE590"
                  cx={18.5}
                  cy={-13.36111111111111}
                  r={30.833333333333336}
                >
                  <View
                    blur={15}
                  />
                </View>
              </View>
            </View>
            <View
              clip={true}
              mask={
                <UniconMask
                  attributeData={
                    Object {
                      "0": "#5CFE9D",
                      "1": "#B8C3B7",
                      "2": Array [
                        Object {
                          "path": "M34 20C34 28.8366 26.8366 36 18 36C9.16344 36 2 28.8366 2 20L2 -3.8147e-06L34 -1.01717e-06V20Z",
                        },
                      ],
                      "3": Array [
                        Object {
                          "fillType": "evenOdd",
                          "path": "M16 8C12.2937 6.62854 9.37146 3.70632 8 0C6.62854 3.70632 3.70632 6.62854 0 8C3.70632 9.37146 6.62854 12.2937 8 16C9.37146 12.2937 12.2937 9.37146 16 8ZM8 11C9.65685 11 11 9.65685 11 8C11 6.34315 9.65685 5 8 5C6.34315 5 5 6.34315 5 8C5 9.65685 6.34315 11 8 11Z",
                        },
                      ],
                    }
                  }
                  overlay={true}
                  showBorder={false}
                  size={32}
                />
              }
            >
              <View
                color="#000000"
                height={32}
                opacity={0.08}
                width={32}
                x={0}
                y={0}
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "height": 12,
              "width": 12,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "display": "flex",
            },
          ]
        }
      >
        <Text
          allowFontScaling={true}
          maxFontSizeMultiplier={1.4}
          style={
            Array [
              Object {
                "color": "#0E1524",
                "display": "flex",
                "fontFamily": "Inter-Medium",
                "fontSize": 17,
                "lineHeight": 24,
              },
            ]
          }
        >
          0xf39F...2266
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexGrow": 0,
            "flexShrink": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#FC72FF",
            "borderRadius": 999999,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "column",
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FC72FF",
              "borderRadius": 999999,
              "display": "flex",
              "height": 8,
              "opacity": 1,
              "width": 8,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
